src/Formatters/FormatterTrait.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Drush\Formatters;
+
+use Consolidation\OutputFormatters\Options\FormatterOptions;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+
+trait FormatterTrait
+{
+    public function addFormatterOptions()
+    {
+        $formatterOptions = new FormatterOptions($this->getConfigurationData(), []);
+        $reflection = new \ReflectionMethod($this, 'doExecute');
+        $inputOptions = $this->formatterManager->automaticOptions($formatterOptions, $reflection->getReturnType()->getName());
+        foreach ($inputOptions as $inputOption) {
+            $mode = $this->getPrivatePropValue($inputOption, 'mode');
+            $suggestedValues = $this->getPrivatePropValue($inputOption, 'suggestedValues');
+            $this->addOption($inputOption->getName(), $inputOption->getShortcut(), $mode, $inputOption->getDescription(), $inputOption->getDefault(), $suggestedValues);
+        }
+    }
+
+    /**
+     * Format the structured data as per user input and the command definition.
+     */
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $configurationData = $this->getConfigurationData($this);
+        $formatterOptions = new FormatterOptions($configurationData, $input->getOptions());
+        $formatterOptions->setInput($input);
+        $data = $this->doExecute($input, $output);
```

Comment:
I was thinking maybe doExecute should be named "fetch()"

Reply:
I think doExecute() better communicates the wrapping we are doing.